### DISCUSSION TITLE
Route aliases from API

### DISCUSSION THREAD

**Discussion by raneio** [nuxt/nuxt#22741] [Questions]
We don't have a prefix before the product group pages, so I have to do it with aliases. How can I add aliases to the page, so that I fetch a list of aliases from the API.

I tried the following but it gives an error: `[Vue warn]: withAsyncContext called without active current instance. This is likely a bug.
[nuxt] [request error] [unhandled] [500] __temp is not defined`

```typescript
definePageMeta({
  name: "Navigation",
  alias: await $fetch("/nuxt-api/groups/").then((groups: any) =>
    groups.map((group: any) => `/${group.slug}`)
  ),
});
```
`/pages/nav.vue`

**Comment by raneio** [comment]
In my case it's probably better to use the path property with ReqExp. That's how I get the url to change when clicking on NuxtLink. But the problem is the same, can I use definePageMeta with data from the API.

```typescript
definePageMeta({
  name: "Navigation",
  path: "/:slug(men|women|kids)+",
});
```

**Comment by raneio** [comment]
No solution has been found for this yet. Any ideas how to solve this? In this way I mentioned or in some other way?

**Comment by or2e** [comment]
@raneio 

What about middleware + addRoue?
https://router.vuejs.org/guide/advanced/dynamic-routing.html#Adding-Routes

**Reply by or2e** [reply]
@raneio 
https://github.com/nuxt/nuxt/issues/23337

**Comment by raneio** [comment]
My solution:

`/plugins/initRoutes.ts`

```
import ContentPage from "~/pages/content/[...slug].vue";

export default defineNuxtPlugin(async (nuxtApp) => {
  const router = nuxtApp.$router as any;
  const routes: any = await $fetch(`/api/routes/`);

  router.addRoute({
    path: `/:slug(${routes.pages.join("|")})+`,
    component: ContentPage,
    name: "content-page-routes",
  });
});
```
